cv++_main.py

```diff
@@ -5,59 +5,83 @@
 import time as t
 import tkinter as tk
 import threading as th
-
+import logging
+    
 class clipboard:
-    def __init__(self):
+    def __init__(self, loglevel):
         self.clipboard = {}
-        self.clipboard_length = 0
+        self.clipboard_logger = logging.getLogger('clipboard')
+        self.clipboard_logger.setLevel(loglevel)
 
-    def add(self, value:str) -> dict:
-        self.clipboard[self.clipboard_length+1] = value
-        self.clipboard_length = self.clipboard_length+1
+        # log to predetermined file
+        self.handler = logging.FileHandler('logs.log')
 
-        return self.clipboard
+        self.formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+        self.handler.setFormatter(self.formatter)
+        self.clipboard_logger.addHandler(self.handler)
 
-    def clear(self) -> dict:
-        self.clipboard = {}
-        self.clipboard_length = 0
+        # track release of keys so that someone can't
+        # overwrite clipboard content by holding
+        # key.
+        self.keybinds = {
+            'copy_down': False,
+            'paste_down': False
+        }
+
+        # store if key was pressed at last loop
+        # to copy when it isn't.
+        self.last_iter = {
+            'copy': False,
+            'paste': False
+        }
+
+    def add(self, value:str, index:int) -> dict:
+        self.clipboard[index] = value
 
         return self.clipboard
 
     def listen_continuously(self) -> None:
         # to copy - CTRL+ALT+NUM
         # compensate for held keys
-        held = {'copy': False,'paste': False}
-        print('listening')
+        self.clipboard_logger.info("listening...")
 
         while True:
             # paste with ctrl+alt+v+{num}
             # check from 1-9. 10 can't be typed as a single key
             # so we use 0 and replace as 10.
-            for i in range(1,9):
+            for i in range(1,11):
                 i2 = 0 if i == 10 else i
 
                 if k.is_pressed(f'ctrl+c+{str(i2)}'):
-                    if not held['copy']:
-                        # 0-9 (10 objects)
-                        if self.clipboard_length >= 10:
-                            self.clear()
-
-                        self.add(pyp.paste())
-                        print('added ' + pyp.paste() + ' to cache at ' + str(self.clipboard_length))
-                    held['copy'] = True
+                    self.keybinds['copy_down'] = True
+                    self.last_iter['copy'] = True
                 else:
-                    held['copy'] = False
+                    self.keybinds['copy_down'] = False
 
                 if k.is_pressed(f'ctrl+alt+{str(i2)}'):
-                    if not held['paste']:
+                    self.keybinds['paste_down'] = True
+                    self.last_iter['paste'] = True
```

Comment:
Same for here, we should add the code to actually copy back the text to a cache.